DeleteLLVMComments.java

```diff
@@ -0,0 +1,56 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.util.Scanner;
+
+// Copyright (c) 2015 K Team. All Rights Reserved.
+public class DeleteLLVMComments {
+
+    /**
+     * @author LiyiLi
+     * This java code takes in a LLVM program and
+     * eliminates the LLVM comments in the program, then
+     * outputs the modified LLVM program
+     * to a temporary LLVM file.
+     */
+    public static void main(String[] args) {
+        String fileName = args[0];
+        File llvmFile = new File(fileName);
+        if(llvmFile.isFile()){
+            try {
+                Scanner scanner = new Scanner(llvmFile);
+                PrintWriter newLlvmProgam = new PrintWriter("llvmtemp.ll");
+                /**
+                 * Declare a variable to indicate whether or
+                 * not the current scanning character is inside
+                 * a double quote block.
+                 */
+                boolean isInQuoteBlock = false;
+                
+                while(scanner.hasNextLine()){
+                    String currentLine = scanner.nextLine();
+                    String newPrintLine = "";
+                    for(int i = 0; i < currentLine.length(); ++i){
+                        if(currentLine.charAt(i) == '"'){
```

Comment:
Can you please add this as a comment in your Java code then?  I would be tempted to think it is a bug in your code without this information.

Grigore

p.s. Sorry for my delayed replies, lots of things going on.  Please arrange your schedule so that you do not depend on me answering immediately to your requests.  Or if very urgent, come to my door.

________________________________
From: Liyi Li [notifications@github.com]
Sent: Sunday, February 15, 2015 3:08 PM
To: kframework/llvm-semantics
Cc: Rosu, Grigore
Subject: Re: [llvm-semantics] Java engine defining more syntax (#8)


In DeleteLLVMComments.java<https://github.com/kframework/llvm-semantics/pull/8#discussion_r24726073>:

> +        if(llvmFile.isFile()){
> +            try {
> +                Scanner scanner = new Scanner(llvmFile);
> +                PrintWriter newLlvmProgam = new PrintWriter("llvmtemp.ll");
> +                /**
> +                 * Declare a variable to indicate whether or
> +                 * not the current scanning character is inside
> +                 * a double quote block.
> +                 */
> +                boolean isInQuoteBlock = false;
> +
> +                while(scanner.hasNextLine()){
> +                    String currentLine = scanner.nextLine();
> +                    String newPrintLine = "";
> +                    for(int i = 0; i < currentLine.length(); ++i){
> +                        if(currentLine.charAt(i) == '"'){


In LLVM, you cannot write \" in any string. The LLVM parser will reject this.

—
Reply to this email directly or view it on GitHub<https://github.com/kframework/llvm-semantics/pull/8/files#r24726073>.